fastwarc/fastwarc/cli.py

```diff
@@ -167,13 +169,87 @@ def check(infile, decompress_alg, verify_payloads, quiet, output):
                     click.echo(rec)
                 sys.exit(1)
 
+@main.command()
+@click.argument('infile', type=click.Path(dir_okay=False, exists=True))
+@click.argument('offset', type=click.INT)
+@click.option('--payload', is_flag=True,
+              help='output only record payload (transfer and/or content encoding are preserved')
+@click.option('--headers', is_flag=True, help='output only record (and HTTP) headers')
+def extract(infile, offset, payload, headers):
+    """Extract WARC record by offset.
+    """
+    compl_record = not (payload or headers)
+    with open(infile, 'rb') as stream:
+        stream.seek(offset)
+        record = next(iter(ArchiveIterator(stream)))
+        if headers or compl_record:
+            ostream = PythonIOStreamAdapter(sys.stdout.buffer)
```

Comment:
Instead of hard-coding `sys.stdout` here, why not add an output file option here as well, similar to who `fastwarc index` works?